htdocs/include/comment_post.php

```diff
@@ -125,73 +131,33 @@
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            ' ',
-            '');
-        $com_user    = trim($_POST['com_user']);
-        $com_user    = $myts->stripSlashesGPC($com_user);
-        $com_user    = $myts->xoopsCodeDecode($com_user);
-        $com_user    = $myts->filterXss($com_user);
-        $com_user    = strip_tags($com_user);
-        $com_user    = strtolower($com_user);
-        $com_user    = htmlentities($com_user, ENT_COMPAT, 'utf-8');
-        $com_user    = preg_replace('`\[.*\]`U', ' ', $com_user);
-        $com_user    = preg_replace('`&(amp;)?#?[a-z0-9]+;`i', ' ', $com_user);
-        $com_user    = preg_replace('`&([a-z])(acute|uml|circ|grave|ring|cedil|slash|tilde|caron|lig);`i', '\\1', $com_user);
-        $com_user    = str_replace($search_arr, $replace_arr, $com_user);
+            '&'
+        );
+        $com_user = Request::getString('com_user', 'POST', '');
+        $com_user = str_replace($search_arr, '', $com_user);
+        //$com_user = strtolower($com_user);
+        $com_user = htmlentities($com_user, ENT_COMPAT, 'utf-8');
 
         // Check Url
-        if (!empty($_POST['com_url'])) {
-            $com_url = trim($_POST['com_url']);
-            $com_url = filter_var($com_url, FILTER_VALIDATE_URL, FILTER_FLAG_SCHEME_REQUIRED);
+        $com_url = Request::getUrl('com_url', '', 'POST');
+        if ('' !== $com_url) {
+            $com_url = filter_var($com_url, FILTER_VALIDATE_URL);
+            if (is_string($com_url) && (false === preg_match("#^https?://#", $com_url))) {
+                $com_url = false;
+            }
+            if (false === $com_url) {
+                $com_url='';
+            }
         }
 
         // Check Email
-        $com_email = $myts->stripSlashesGPC(trim($_POST['com_email']));
-        $com_email = htmlspecialchars(trim($com_email), ENT_QUOTES);
-        $com_email = filter_var($com_email, FILTER_VALIDATE_EMAIL);
+        $com_email = Request::getEmail('com_email', '', 'POST');
         // Invalid email address
-        if (!checkEmail($com_email)) {
+        if (empty($com_email)) {
             $error_message .= _US_INVALIDMAIL . '<br>';
         }
-        if (strrpos($com_email, ' ') > 0) {
-            $error_message .= _US_EMAILNOSPACES . '<br>';
-        }
         // Check forbidden email address if current operator is not an administrator
-        if (!$xoopsUser_isAdmin) {
+        if (!(is_object($xoopsUser) && $xoopsUser->isAdmin())) {
```

Comment:
Should we make the same change everywhere, or is it only a special case here? Just curious...

Reply:
From what I can see, the variable `$xoopsUser_isAdmin` is local to `\XoopsUserUtility::validate()`, so using it here was a definite bug.